velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -47,12 +47,18 @@ ReaderBase::ReaderBase(
 }
 
 void ReaderBase::loadFileMetaData() {
-  bool preloadFile_ = fileLength_ <= filePreloadThreshold_;
+  preloadFile_ = fileLength_ <= filePreloadThreshold_ ||
+      fileLength_ <= directorySizeGuess_;
```

Comment:
also, why include `fileLength_ <= directorySizeGuess_`?

Reply:
@zhli1142015 @JkSelf Can you help to answer this question? Thanks!

Reply:
@majetideepak Because in the next line, `readSize` will be `fileLength_` if `fileLength_ <= directorySizeGuess_`. In this case the whole file is loaded.
```
  uint64_t readSize =
      preloadFile_ ? fileLength_ : std::min(fileLength_, directorySizeGuess_);
```
While the code looks weird. Optimized this part of logic in the latest commit.